grails-app/services/org/pih/warehouse/invoice/PrepaymentInvoiceService.groovy

```diff
@@ -182,7 +182,13 @@ class PrepaymentInvoiceService {
             return null
         }
         InvoiceItem inverseItem = createFromShipmentItem(shipmentItem)
-        Integer quantity = invoiceItem.quantity >= quantityInverseable ? quantityInverseable : invoiceItem.quantity
+        Integer quantity
+        if (orderItem.isCompletelyFulfilled() && orderItem.isFullyInvoiced()) {
+            // If quantity if fully shipped and fully invoiced set full inverse quantity
+            quantity = quantityInverseable
+        } else {
+            quantity = invoiceItem.quantity >= quantityInverseable ? quantityInverseable : invoiceItem.quantity
+        }
```

Comment:
In the code above, we use returns instead of else, so maybe to keep the code consistent, it would be better to use the same approach here?

+ What do you think about writing some simple unit tests? :sadsmile:

Reply:
In the code above we return nulls and we don't want to execute code that is below. Here we want to determine the quantity that is going to be used in the latter part of the code

Reply:
> + What do you think about writing some simple unit tests? :sadsmile:

When it is QA'ed we can think about bumping up test coverage in entire partial invoice feature

Reply:
> When it is QA'ed we can think about bumping up test coverage in entire partial invoice feature

good idea, but we can't forget about that as always